web/components/feed/contract-activity.tsx

```diff
@@ -28,38 +28,40 @@ export function ContractActivity(props: {
   className?: string
   betRowClassName?: string
 }) {
-  const { user, mode, contractPath, className, betRowClassName } = props
+  const { user, mode, contractPath, className, betRowClassName, contract } =
+    props
 
-  const contract = useContract(props.contract.id) ?? props.contract
+  const liveContract = useContractWithPreload(contract) ?? props.contract
```

Comment:
This only makes sense if useContract is not getting updates, but I thought it would be. Could be a problem with how I'm using the `react-query` library...

In any case, happy to proceed if you think this solves the issue.

Reply:
Yes I did a/b testing and this does fix the issue. I suppose ListenForValues and/or onSnapshot works slightly differently than useFirestoreDocumentData?